### PR TITLE
POC for a mobile WeclomeView

### PR DESCRIPTION
**Pull request by akuleshov7**

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import react.router.Route
+import react.router.Routes
+
+/**
+ * Just put a map: View -> Route URL to this list
+ */
+val mobileRoutes = VFC {
+    Routes {
+        listOf(
+            AboutUsMobileView::class.react.create() to FrontendRoutes.ABOUT_US.path,
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt:22]: @petertrr @nulls @sanyavertolet  if you will have any ideas of how to cast it, so I won't need to have `.react.create()`, will be great. Simple map does not work, due to receiver type...

```
.map { it.react.create() }
```

Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt:22]: migrate to `FC`

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/MarketingTitles.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import react.ChildrenBuilder
+import react.dom.html.ReactHTML
+
+/**
+ * @param textColor
+ * @param isDark
+ */
+fun ChildrenBuilder.welcomeMarketingTitle(textColor: String, isDark: Boolean = false) {
+    ReactHTML.div {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/MarketingTitles.kt:20]:
```suggestion
    div {
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/MarketingTitles.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        className = ClassName("col-lg-4 ml-auto mt-3 mb-5 mr-5 ml-0 $textColor")
+        marketingTitle("Software", isDark)
+        marketingTitle("Analysis", isDark)
+        marketingTitle("Verification &", isDark)
+        marketingTitle("Evaluation", isDark)
+        ReactHTML.h3 {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/MarketingTitles.kt:26]:
```suggestion
        h3 {
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/pagers/WelcomePager.kt [hunk 4]
```diff
@@ -4,14 +4,15 @@ import com.saveourtool.save.frontend.externals.animations.Animation
 import react.ChildrenBuilder
 
 val allWelcomePagers = listOf(
-    listOf(HighLevelSave),
+    // listOf(HighLevelSave),
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/pagers/WelcomePager.kt:7]: Should it be commented?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt:22]: Maybe wrap it into `VFC`s? In that way it seems possible to easily iterate over the list and create components

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt:22]: > migrate to `FC`

It seems to be a long journey, can we do it now?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/pagers/WelcomePager.kt (hunk 4 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/welcome/pagers/WelcomePager.kt:7]: yeah, will think of how to draw it - in the next PR

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt (hunk 1 above)
Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt:22]: > Maybe wrap it into `VFC`s? In that way it seems possible to easily iterate over the list and create components

wrapper looks much more ugly, I want to remove existing actually

Comment (petertrr) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/routing/MobileRouting.kt:22]: > Simple map does not work, due to receiver type...

What type is inferred in this case? `.react` is an extension on `KClass<out Component<P, *>>`, maybe list can be cast somehow?

**Pull request merged**